### PR TITLE
Add ability to create variable length'd plot lines

### PR DESCRIPTION
**Pull request by okcoker**
As of right now plot lines extend the height (or width?) of the chart. In my case I want to be able to control  how high the plot lines rise. This simple addition will allow much greater control of that.

An example using this change would be something like:

``` js
plotLines:  [{
    value: i,
    width: 1,
    verticalAlign: 'middle',
    color: '#ffffff',
    formatter(path) {
        // path will be in the format of ["M", 108.5, 10, "L", 108.5, 186]
        path[2] = 50 * this.value; // Edit height of lines
        return path; // return new path array
     }
})

```

### CODE REVIEW

File: js/parts/PlotLineOrBand.js [hunk 1]
```diff
@@ -54,6 +55,11 @@ Highcharts.PlotLineOrBand.prototype = {
 		// plot line
 		if (width) {
 			path = axis.getPlotLinePath(value, width);
+
+			if (typeof formatter === 'function') {
+				path = formatter(path);
```
Comment (okcoker) [inline js/parts/PlotLineOrBand.js]: This should probably be `path = formatter.call(context, path)`. Just not sure what to set the context to.

**Comment by ericmulder** [comment]
Very nice solution, I use it in my project!

**Comment by JulianLeviston** [comment]
Would love to have this. Sadly we're not on latest highcharts.

**Comment by TorsteinHonsi** [comment]
You can do something similar by wrapping the `PlotBandOrLine.prototype.render` function : http://jsfiddle.net/highcharts/p7frp7o6/

**Pull request closed without merging**